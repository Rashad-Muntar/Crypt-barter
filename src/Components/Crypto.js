import React from 'react';
import PropTypes from 'prop-types';
import '../Assets/crypto.css';

const Crypto = ({
  name, price, icon, perChange, rank, symbol,
}) => (
  <div className="card-wrapper">
    <div className="icon-section">
      <p className="header">{name}</p>
      <img src={icon} alt="icon" />
    </div>
    <div className="price-area">
      <p className="price">
        â‚¬
        {price.toFixed(4)}
      </p>
      { perChange < 1 ? (
        <div className="low">
          {perChange}
          %
          <i className="arrow down icon" />
        </div>
      ) : (
        <div className="high">
          {perChange}
          %
          <i className="arrow up icon" />
        </div>
      )}
    </div>
    <div className="symbol-area">
      <p className="rank">
        rank:
        {rank}
      </p>
      <p className="symbol">{symbol}</p>
    </div>
  </div>
);

Crypto.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  perChange: PropTypes.string.isRequired,
  rank: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
};

export default Crypto;
