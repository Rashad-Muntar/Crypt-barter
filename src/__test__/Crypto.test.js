import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Crypto from '../Components/Crypto';

describe('StockCard DOM', () => {
  it('renders correctly', () => {
    const cryptoCard = renderer.create(
      <Crypto
        name="name"
        price={5}
        icon="df"
        rank="f"
        perChange="3"
        symbol="34"
      />,
    ).toJSON();
    expect(cryptoCard).toMatchSnapshot();
  });
});

// it('renders correctly when store is provided', () => {
//   const store = createStore(rootReducer);
//   const cryptoList = renderer.create(
//     <Provider store={store}>
//       <Crypto />
//     </Provider>,
//   ).toJSON();
//   expect(cryptoList).toMatchSnapshot();
// });
