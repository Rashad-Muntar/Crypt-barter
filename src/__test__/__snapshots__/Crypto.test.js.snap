// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockCard DOM renders correctly 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="icon-section"
  >
    <p
      className="header"
    >
      name
    </p>
    <img
      alt="icon"
      src="df"
    />
  </div>
  <div
    className="price-area"
  >
    <p
      className="price"
    >
      â‚¬
      5.0000
    </p>
    <div
      className="high"
    >
      3
      %
      <i
        className="arrow up icon"
      />
    </div>
  </div>
  <div
    className="symbol-area"
  >
    <p
      className="rank"
    >
      rank:
      f
    </p>
    <p
      className="symbol"
    >
      34
    </p>
  </div>
</div>
`;
